



<span id="top"></span>




# <span dir="auto">GMOD Evo Hackathon</span>









<table style="vertical-align: middle; border: 2px solid #A6A6BC"
data-cellpadding="10">
<colgroup>
<col style="width: 50%" />
<col style="width: 50%" />
</colgroup>
<tbody>
<tr class="odd">
<td><a href="http://www.nescent.org/wg_gmodevohackathon/" rel="nofollow"
title="GMOD Evo Hackathon"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/a/a8/EvoHackathonLogo.png/200px-EvoHackathonLogo.png"
srcset="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/a/a8/EvoHackathonLogo.png/300px-EvoHackathonLogo.png 1.5x, https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/a/a8/EvoHackathonLogo.png/400px-EvoHackathonLogo.png 2x"
width="200" height="200" alt="GMOD Evo Hackathon" /></a></td>
<td><span style="font-size: 180%; line-height: 120%"><strong><a
href="http://www.nescent.org/wg_gmodevohackathon/" class="external text"
rel="nofollow">Tools for Evolutionary Biology
Hackathon</a></strong><br />
November 8-12, 2010<br />
<a href="http://nescent.org/" class="external text"
rel="nofollow">NESCent</a>, Durham, North Carolina, USA</span><br />
<br />
<a href="File:EvoHackRoom2010.jpg" class="image"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/a/af/EvoHackRoom2010.jpg/290px-EvoHackRoom2010.jpg"
srcset="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/a/af/EvoHackRoom2010.jpg 1.5x, https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/a/af/EvoHackRoom2010.jpg 2x"
width="290" height="100" alt="EvoHackRoom2010.jpg" /></a> <a
href="File:EvoHackWhiteboard2010.jpg" class="image"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/f/f0/EvoHackWhiteboard2010.jpg/149px-EvoHackWhiteboard2010.jpg"
srcset="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/f/f0/EvoHackWhiteboard2010.jpg/224px-EvoHackWhiteboard2010.jpg 1.5x, https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/f/f0/EvoHackWhiteboard2010.jpg/298px-EvoHackWhiteboard2010.jpg 2x"
width="149" height="100" alt="EvoHackWhiteboard2010.jpg" /></a></td>
</tr>
</tbody>
</table>

GMOD held a hackathon November 8-12, 2010, at the
<a href="http://nescent.org" class="external text"
rel="nofollow">National Evolutionary Synthesis Center (NESCent)</a> in
Durham, North Carolina. This hackathon focused on improving GMOD's
support for [evolutionary
biology](Category%3AEvolution "Category%3AEvolution").

The [Open Call for
Participation](GMOD_Evo_Hackathon_Open_Call "GMOD Evo Hackathon Open Call")
went out on August 1, 2010, and remained open until August 25.
Participants have been selected and notified of their status.


## Contents



- [<span class="tocnumber">1</span>
  <span class="toctext">Synopsis</span>](#Synopsis)
  - [<span class="tocnumber">1.1</span>
    <span class="toctext">Subgroups</span>](#Subgroups)
  - [<span class="tocnumber">1.2</span> <span class="toctext">PAG
    Poster</span>](#PAG_Poster)
- [<span class="tocnumber">2</span>
  <span class="toctext">Background</span>](#Background)
- [<span class="tocnumber">3</span> <span class="toctext">Specific
  objectives</span>](#Specific_objectives)
  - [<span class="tocnumber">3.1</span> <span class="toctext">Viewing
    tools for comparative genomics
    data</span>](#Viewing_tools_for_comparative_genomics_data)
  - [<span class="tocnumber">3.2</span>
    <span class="toctext">Visualization of phylogenetic data and
    trees</span>](#Visualization_of_phylogenetic_data_and_trees)
  - [<span class="tocnumber">3.3</span> <span class="toctext">Population
    Diversity and Phenotype
    support</span>](#Population_Diversity_and_Phenotype_support)
- [<span class="tocnumber">4</span> <span class="toctext">Hackathon
  Structure</span>](#Hackathon_Structure)
  - [<span class="tocnumber">4.1</span> <span class="toctext">Before the
    Event</span>](#Before_the_Event)
  - [<span class="tocnumber">4.2</span> <span class="toctext">During the
    Event</span>](#During_the_Event)
  - [<span class="tocnumber">4.3</span>
    <span class="toctext">Deliverables / Event
    Results</span>](#Deliverables_.2F_Event_Results)
- [<span class="tocnumber">5</span> <span class="toctext">Participant
  Funding</span>](#Participant_Funding)
- [<span class="tocnumber">6</span> <span class="toctext">March 2011
  Satellite</span>](#March_2011_Satellite)
- [<span class="tocnumber">7</span>
  <span class="toctext">Timeline</span>](#Timeline)
- [<span class="tocnumber">8</span>
  <span class="toctext">Sponsorship</span>](#Sponsorship)
- [<span class="tocnumber">9</span> <span class="toctext">Organizing
  Committee</span>](#Organizing_Committee)



# <span id="Synopsis" class="mw-headline">Synopsis</span>

This hackathon addressed critical gaps in the capabilities of the
Generic Model Organism Database (GMOD) toolbox that limited its utility
for evolutionary research. Specifically, we [focused
on](#Specific_Objectives) tools for 1) viewing comparative genomics
data; 2) visualizing phylogenomic data; and 3) supporting population
diversity data and phenotype annotation.

The event brought together a group of 30 software developers, end-user
representatives, and documentation experts who would otherwise not have
met. The participants included key developers of GMOD components that
lacked features critical for emerging evolutionary biology research,
developers of informatics tools in evolutionary research that lacked
GMOD integration, and informatics-savvy biologists who represented
end-user requirements.

This hackathon provided a unique opportunity to infuse the community of
GMOD developers with a heightened awareness of unmet needs in
evolutionary biology that GMOD components have the potential to fill,
and for tool developers in evolutionary biology to better understand how
best to extend or integrate with already existing GMOD components.

## <span id="Subgroups" class="mw-headline">Subgroups</span>

*Due to the closure of NESCent, the links below to the hackathon wiki
originally hosted by NESCent no longer resolve. A
<a href="http://dx.doi.org/10.5281/zenodo.19020" class="external text"
rel="nofollow">permanent archive of the wiki</a> in the form of a
MediaWiki export is available at Zenodo. To view the content of that
wiki, one first needs to reinstantiate a MediaWiki instance from the
export.*



<a href="File:EvoHackLaptops2010.jpg" class="image"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/7/7f/EvoHackLaptops2010.jpg/200px-EvoHackLaptops2010.jpg"
srcset="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/7/7f/EvoHackLaptops2010.jpg/300px-EvoHackLaptops2010.jpg 1.5x, https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/7/7f/EvoHackLaptops2010.jpg 2x"
width="200" height="119" alt="EvoHackLaptops2010.jpg" /></a>



The <a href="https://www.nescent.org/wg_gmodevohackathon/Participants"
class="external text" rel="nofollow">30 participants</a> self-organized
into eight groups with at least one group addressing each of the event's
three
[objectives](GMOD_Evo_Hackathon#Specific_objectives "GMOD Evo Hackathon").
The outcomes for each group are summarized below.

<a href="https://www.nescent.org/wg_gmodevohackathon/GMatchbox"
class="external text" rel="nofollow">GMatchbox</a>  
Worked on establishing a common database backend and
[JSON](Glossary#JSON "Glossary")-based API for comparative genomics
data, using several visualization tools (including
[JBrowse](JBrowse.1 "JBrowse") and
[GBrowse_syn](GBrowse_syn.1 "GBrowse syn")) as targets. Will enable
sharing of comparative data in multiple tools from multiple sources.

<!-- -->

<a href="https://www.nescent.org/wg_gmodevohackathon/GBrowse_syn2"
class="external text" rel="nofollow">GBrowse_syn2</a>  
[GBrowse_syn](GBrowse_syn.1 "GBrowse syn") is built on and takes
advantage of the [GBrowse](GBrowse.1 "GBrowse") genome browser code and
config files. However, it did not work well with GBrowse2, due to
significant architectural changes. This group refactored GBrowse2 to
naturally support GBrowse_syn. This work will also enable several other
GBrowse1-only applications ([SynView](SynView "SynView"), [Primer
Designer](PrimerDesigner.pm "PrimerDesigner.pm"), ...) to be ported to
GBrowse2 as well. Two participants also became core GBrowse_syn
developers.

<!-- -->

<a href="https://www.nescent.org/wg_gmodevohackathon/JBrowse_syn"
class="external text" rel="nofollow">JBrowse_syn</a>  
This group set out to extend [JBrowse](JBrowse.1 "JBrowse") to be a
comparative genomics browser. The group removed the existing "single
genome" assumption from the code and successfully displayed several
genomes in parallel. Several participants also became familiar with
JBrowse code and architecture.

<!-- -->

<a href="https://www.nescent.org/wg_gmodevohackathon/PhyloBox"
class="external text" rel="nofollow">PhyloBox</a>  
<a href="http://phylobox.appspot.com/" class="external text"
rel="nofollow">PhyloBox</a> is a flexible and fast web based tree
visualization program. At the hackathon the PhyloBox team extended
PhyloBox in numerous ways to make it a "widget" that can interact with
other widgets. PhyloBox documentation was also created.

<!-- -->

<a
href="https://www.nescent.org/wg_gmodevohackathon/Comparative_Genomics_Visualization:Phylobox_JBrowse"
class="external text" rel="nofollow">Integration PhyloBox JBrowse
Integration</a>  
The group is the perfect example of the interaction that can happen at a
hackathon. They worked with the
<a href="https://www.nescent.org/wg_gmodevohackathon/JBrowse_syn"
class="external text" rel="nofollow">JBrowse_syn</a>,
<a href="https://www.nescent.org/wg_gmodevohackathon/PhyloBox"
class="external text" rel="nofollow">PhyloBox</a> and
<a href="https://www.nescent.org/wg_gmodevohackathon/GMatchBox"
class="external text" rel="nofollow">GMatchbox</a> groups to enable
integration of these three technologies. This group was very helpful at
getting teams to work together.

<!-- -->

<a
href="https://www.nescent.org/wg_gmodevohackathon/Natural_Diversity_and_Phenotypes_Subgroup"
class="external text" rel="nofollow">Natural Diversity and Phenotypes in
Chado</a>  
This group focused on two outcomes, both relating to
<a href="Chado" class="mw-redirect" title="Chado">Chado</a>. The first
was a prototype Rails application that provided a web interface to the
new [Natural
Diversity](Chado_Natural_Diversity_Module.1 "Chado Natural Diversity Module")
module in Chado. This was built on top of the emerging [Chado on
Rails](Chado_on_Rails "Chado on Rails") project. The second was a better
understanding, slight refactoring, and updated documentation for Chado's
phenotype module.

<!-- -->

<a
href="https://www.nescent.org/wg_gmodevohackathon/Galaxy_and_HyPhy_Subgroup"
class="external text" rel="nofollow">Galaxy + HyPhy</a>  
[Galaxy](Galaxy.1 "Galaxy") is both a workflow system and a means of
persisting computational pipelines and results. This group worked on
improving Galaxy's ability to integrate interactive tools, using
<a href="http://hyphy.org" class="external text"
rel="nofollow">HyPhy</a> as the prototype application. The Galaxy and
HyPhy code bases were modified to support this.

<!-- -->

<a
href="https://www.nescent.org/wg_gmodevohackathon/BioPerl_Enhancements_Subgroup"
class="external text" rel="nofollow">BioPerl</a>  
This subgroup worked on improving tree handling in
[BioPerl](BioPerl "BioPerl"). Specifically, they addressed the handling
of very large trees or large numbers of small trees. BioPerl now
supports storing such trees in a lightweight database instead of in
memory.

## <span id="PAG_Poster" class="mw-headline">PAG Poster</span>

[Robert Buels](User%3ARobertBuels "User%3ARobertBuels") gave a <a
href="http://www.nescent.org/wg/gmodevohackathon/images/3/3f/Gmod_evohack_poster.pdf"
class="external text" rel="nofollow">poster on the hackathon</a> at [PAG
2011](PAG_2011 "PAG 2011").

# <span id="Background" class="mw-headline">Background</span>

The GMOD project is a confederation of intercompatible open-source
projects developing software tools for storing, managing, curating, and
publishing biological data. Although the GMOD project originated from
the goal of developing a generic tool set for common needs among model
organism databases, GMOD tools are meanwhile used by many large and
small, collaborative and single-investigator biological database
projects for the dissemination of experimental results and curated
knowledge.

GMOD's software tools provide a powerful and feature-rich basis for
working with biological, in particular genomic and other molecular data.
However, due to GMOD's historical emphasis on single-genome projects
many GMOD tools still lack features that are critical to effectively
support the comparative, phylogenetic, and natural diversity-oriented
questions frequently asked in evolutionary research.

Recent developments have given rise to a window of opportunity for
forging collaborations towards filling this gap. In particular, the cost
of collecting comparative molecular data on a large or even genomic
scale has recently dropped dramatically, primarily thanks to
next-generation high-throughput sequencing technologies. This has
enabled evolutionary researchers to bring genome-scale molecular data to
bear on key evolutionary questions. It has also allowed single
organism-focused molecular biology labs, who represent GMOD's
traditional user base, to broaden out to multi-organism comparative
approaches. Bringing these two communities with increasingly shared
interests and complementary scientific and technical expertise together
offers an opportunity to start filling GMOD's gaps in these areas while
building on its existing strengths. In addition, such direct interaction
will heighten future awareness of needs of evolutionary researchers
among GMOD developers who have so far mostly supported its traditional
user base, and can in the long term increase the ranks of GMOD
contributors from a field it was not originally designed to serve.

The hackathon format is ideally suited to realize this opportunity. Its
strengths lie in facilitating face-to-face interaction among people with
complementary expertise, and collaborative work on tangible products
that can form the basis of continued partnerships long beyond the end of
the meeting.

# <span id="Specific_objectives" class="mw-headline">Specific objectives</span>

Organizers identified the following broad themes for focusing work at
the event. Before and at the hackathon, the participants refined and
distilled these and other options into concrete implementation targets.
The participants developed criteria for prioritization, such as maturity
of a target for implementation, availability of test data, and potential
for completing or making significant progress towards the target during
the hackathon. Further ideas and discussion topics can be found on the
[Supplemental
Information](GMOD_Evo_Hackathon_Proposal_Supplemental_Information "GMOD Evo Hackathon Proposal Supplemental Information")
page.

## <span id="Viewing_tools_for_comparative_genomics_data" class="mw-headline">Viewing tools for comparative genomics data</span>

[GBrowse_syn](GBrowse_syn.1 "GBrowse syn") is a popular GMOD component
for viewing comparative genomics data, particularly for viewing synteny
between genomes. It does not currently support the [next-generation
sequencing (NGS)
data](Next_Generation_Sequencing "Next Generation Sequencing")
increasingly available for comparative genomics and emerging model
systems. Support for NGS data was identified by the EMS working group as
a high priority.

In particular, GBrowse_syn lacks support for the
<a href="http://samtools.sourceforge.net/" class="external text"
rel="nofollow">Sequence Alignment Format (SAM)</a>, its mechanism of
storing genome comparisons does not scale beyond a few organisms, and
the means for tracking the necessary alignment metadata in
<a href="Chado" class="mw-redirect" title="Chado">Chado</a> are
insufficient.

In addition to filling those gaps, GBrowse_syn would also particularly
stand to benefit from the event by gaining a more sustainable developer
base.

## <span id="Visualization_of_phylogenetic_data_and_trees" class="mw-headline">Visualization of phylogenetic data and trees</span>

The [GMOD toolkit](GMOD_Components "GMOD Components") at present does
not include web-based alignment viewers, nor can the increasingly
popular [JBrowse](JBrowse.1 "JBrowse") genome browser (the designated
successor of [GBrowse](GBrowse.1 "GBrowse")) display multiple sequence
alignments. GMOD also lacks a phylogenetic tree widget.

Implementing these from scratch would be far beyond a suitable hackathon
target. However, <a href="http://solgenomics.net" class="external text"
rel="nofollow">SGN</a> has a relatively mature
<a href="http://solgenomics.net/tools/align_viewer/"
class="external text" rel="nofollow">web-based multiple alignment and
tree browser</a> that could be extracted from SGN's codebase and
transformed into a GMOD component, an add-on for JBrowse. Current
Java-based tree viewers (such as
<a href="http://www.phylosoft.org/atv/" class="external text"
rel="nofollow">Archaeopteryx</a> or
<a href="http://www.phylowidget.org" class="external text"
rel="nofollow">PhyloWidget</a>) could be used as the basis for a
JavaScript-based tree viewer (or an applet that can be controlled
through JavaScript) that integrates with JBrowse.

## <span id="Population_Diversity_and_Phenotype_support" class="mw-headline">Population Diversity and Phenotype support</span>

GMOD's capabilities in managing phenotype and natural diversity data is
scattered across partially redundant and outdated modules, does not
support modern ontology-based entity-quality data, and lacks a
web-interface. The sophisticated phenotype annotation tools that do
exist cannot interface with Chado, GMOD's central relational data model.
Yet, phenotypic and genetic diversity data are central to many
evolutionary research questions.

A [Natural Diversity Module
initiative](Chado_Natural_Diversity_Module_Working_Group "Chado Natural Diversity Module Working Group")
to address at least the deficiencies within Chado has already formed
earlier this year. Several key developers (one of the original
developers of the module, and the developer of Phenex, a phenotype
curation tool) are already local to NESCent, and so the hackathon
provides a unique opportunity to review and refine the natural diversity
data model face-to-face, and to integrate it with an updated and
reconciled phenotype module. A recently reported prototype of a Chado
data adapter for Phenote, GMODs phenotype annotation tool, could be
generalized to become the data persistence interface for such data.

Aside from the data model deficiencies, the
<a href="http://aniseed-ibdm.univ-mrs.fr/" class="external text"
rel="nofollow">ANISEED</a> project has started efforts to generalize its
sophisticated atlas/image-based web interface for phenotype data, and to
make it operate on top of Chado. The hackathon could harness this
synergy to help this effort leap forward, which could ultimately provide
GMOD with the currently missing web-interface for such data.

# <span id="Hackathon_Structure" class="mw-headline">Hackathon Structure</span>

## <span id="Before_the_Event" class="mw-headline">Before the Event</span>

Discussion of ideas and sometimes even design actually starts well
before the hackathon, on mailing lists, wiki pages, and conference calls
set up among accepted attendees. This advance work lays the foundation
for participants to be productive from the very first day. This also
means that participants should be willing to contribute some time in
advance of the hackathon itself to participate in this preparatory
discussion.

## <span id="During_the_Event" class="mw-headline">During the Event</span>

Typically, hackathon participants use the morning of the first day of
the event to organize themselves into working groups of between 3 and 6
people, each with a focused implementation objective. Ideas and
objectives are discussed, and attendees coalesce around the projects in
which they have the most experience or interest.

## <span id="Deliverables_.2F_Event_Results" class="mw-headline">Deliverables / Event Results</span>

The hackathon will use a
<a href="http://www.nescent.org/wg_gmodevohackathon/Main_Page"
class="external text" rel="nofollow">wiki hosted at NESCent</a> during
the event. Once the hackathon is done, relevant content will be copied
from the NESCent wiki to the [GMOD wiki](Main_Page "Main Page"). Each
working group during the event will typically have its own wiki page,
linked from the main hackathon wiki page, where it documents its minutes
and design notes, and provides links to the code and documentation it
produces. Also, since GMOD and NESCent are both committed to open source
principles, all code and documentation produced by participants during
the hackathon must be published under an
<a href="http://www.opensource.org/licenses/category"
class="external text" rel="nofollow">OSI-approved open source
license</a>. As contributions to existing GMOD tools, all hackathon
products will most likely satisfy this requirement automatically.

# <span id="Participant_Funding" class="mw-headline">Participant Funding</span>

NESCent is [sponsoring](#Sponsorship) this hackathon, and had made funds
available to defray costs for qualified participants.

# <span id="March_2011_Satellite" class="mw-headline">March 2011 Satellite</span>


[<img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/8/8e/2011GMODAmericasSatellitesLogo.png/300px-2011GMODAmericasSatellitesLogo.png"
srcset="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/8/8e/2011GMODAmericasSatellitesLogo.png/450px-2011GMODAmericasSatellitesLogo.png 1.5x, https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/8/8e/2011GMODAmericasSatellitesLogo.png/600px-2011GMODAmericasSatellitesLogo.png 2x"
width="300" height="119"
alt="Satellite Meetings at GMOD Americas 2011" />](Satellite_Meetings_-_GMOD_Americas_2011 "Satellite Meetings at GMOD Americas 2011")


We are planning a followup gathering as a [Satellite
Meeting](Satellite_Meetings_-_GMOD_Americas_2011 "Satellite Meetings - GMOD Americas 2011")
at [GMOD Americas 2011](GMOD_Americas_2011 "GMOD Americas 2011"), in
March at NESCent. If you are interested in participating, please add
your name below.



*You do not need to have attended the original hackathon or plan on
attending any other [GMOD Americas
2011](GMOD_Americas_2011 "GMOD Americas 2011") events to participate in
this satellite (or any other satellite). If you have an interest in
extending GMOD and will be in the area or at GMOD Americas 2011, then
you are strongly encouraged to participate.*



  

| Name | Email | Particular Interest? |
|----|----|----|
| Duke Leto (Organizer) | jonathan at leto net |  |
| Hilmar Lapp | hlapp at nescent.org | Large trees in BioPerl cleanup |
|  |  |  |
|  |  |  |
|  |  |  |
| ↑ Add your name and details above |  |  |

# <span id="Timeline" class="mw-headline">Timeline</span>

|  |  |
|---:|----|
| **June 3, 2010** | [Proposal](GMOD_Evo_Hackathon_Proposal "GMOD Evo Hackathon Proposal") submitted to <a href="http://nesscent.org/" class="external text"
rel="nofollow">NESCent</a> |
| **June 10, 2010** | Funding approved |
| **August 1, 2010** | [Open call for participants](GMOD_Evo_Hackathon_Open_Call "GMOD Evo Hackathon Open Call"), applications open |
| **August 25, 2010** | [Open call application](GMOD_Evo_Hackathon_Open_Call "GMOD Evo Hackathon Open Call") deadline |
| **September 16, 2010** | Applicants notified |
| **September 24, 2010** | Deadline for participant attendance commitment |
| **November 8-12, 2010** | Hackathon at NESCent |
| **March 7 (+ ?), 2011** | [Hackathon followup gathering](#March_2011_Satellite) at NESCent as part of [GMOD Americas 2011](GMOD_Americas_2011 "GMOD Americas 2011"). |

# <span id="Sponsorship" class="mw-headline">Sponsorship</span>



<a href="http://nescent.org/" rel="nofollow" title="NESCent"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/3/3a/NESCentLogo152.jpg/120px-NESCentLogo152.jpg"
srcset="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/3/3a/NESCentLogo152.jpg 1.5x, https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/3/3a/NESCentLogo152.jpg 2x"
width="120" height="111" alt="NESCent" /></a>



This event is sponsored by the
<a href="http://www.nescent.org/" class="external text"
rel="nofollow">US National Evolutionary Synthesis Center (NESCent)</a>
through its <a href="http://www.nescent.org/informatics/whitepapers.php"
class="external text" rel="nofollow">Informatics Whitepapers program</a>.
NESCent <a href="http://www.nescent.org/science/proposals.php"
class="external text" rel="nofollow">promotes the synthesis of
information, concepts and knowledge</a> to address significant,
emerging, or novel questions in evolutionary science and its
applications. NESCent achieves this by supporting research and education
across disciplinary, institutional, geographic, and demographic
boundaries.

# <span id="Organizing_Committee" class="mw-headline">Organizing Committee</span>

- [Nicole Washington](User%3ANLWashington "User%3ANLWashington"), chair
- [Sheldon McKay](User%3AMckays "User%3AMckays")
- [Scott Cain](User%3AScott "User%3AScott")
- <a href="User%3ARBuels" class="mw-redirect" title="User%3ARBuels">Robert
  Buels</a>
- <a href="http://informatics.nescent.org/wiki/User%3AHlapp"
  class="external text" rel="nofollow">Hilmar Lapp</a>
- [Dave Clements](User%3AClements "User%3AClements")




[Categories](Special%3ACategories "Special%3ACategories"):

- [Evolution](Category%3AEvolution "Category%3AEvolution")
- [GMOD Developers](Category%3AGMOD_Developers "Category%3AGMOD Developers")
- [GMOD Evo
  Hackathon](Category%3AGMOD_Evo_Hackathon "Category%3AGMOD Evo Hackathon")
- [Events](Category%3AEvents "Category%3AEvents")






## Navigation menu






### 



<a href="Main_Page"
style="background-image: url(../images/GMOD-cogs.png);"
title="Visit the main page"></a>


### Navigation



- <span id="n-GMOD-Home">[GMOD Home](Main_Page)</span>
- <span id="n-Software">[Software](GMOD_Components)</span>
- <span id="n-Categories-.2F-Tags">[Categories /
  Tags](Categories)</span>




### Documentation



- <span id="n-Overview">[Overview](Overview)</span>
- <span id="n-FAQs">[FAQs](Category%3AFAQ)</span>
- <span id="n-HOWTOs">[HOWTOs](Category%3AHOWTO)</span>
- <span id="n-Glossary">[Glossary](Glossary)</span>




### Community



- <span id="n-GMOD-News">[GMOD News](GMOD_News)</span>
- <span id="n-Training-.2F-Outreach">[Training /
  Outreach](Training_and_Outreach)</span>
- <span id="n-Support">[Support](Support)</span>
- <span id="n-GMOD-Promotion">[GMOD Promotion](GMOD_Promotion)</span>
- <span id="n-Meetings">[Meetings](Meetings)</span>
- <span id="n-Calendar">[Calendar](Calendar)</span>




### Tools

- <span id="t-smwbrowselink"><a href="Special%3ABrowse/GMOD_Evo_Hackathon" rel="smw-browse">Browse
  properties</a></span>



- <span id="footer-info-lastmod">Last updated at 13:22 on 13 November
  2015.</span>
<!-- - <span id="footer-info-viewcount">288,487 page views.</span> -->
- <span id="footer-info-copyright">Content is available under
  <a href="http://www.gnu.org/licenses/fdl-1.3.html" class="external"
  rel="nofollow">a GNU Free Documentation License</a> unless otherwise
  noted.</span>

<!-- -->

- <span id="footer-places-about">[About
  GMOD](GMOD%3AAbout "GMOD%3AAbout")</span>

<!-- -->




