



<span id="top"></span>




# <span dir="auto">November 2007 GMOD Meeting</span>









GMOD's November 2007 meeting was held November 5, 1:30PM to November 7,
12:00PM at
<a href="http://www.cshl.edu/" class="external text" rel="nofollow">Cold
Spring Harbor Laboratory</a> following the
<a href="http://meetings.cshl.edu/meetings/info07.shtml"
class="external text" rel="nofollow">Genome Informatics</a> meeting.




## Contents



- [<span class="tocnumber">1</span> <span class="toctext">Pre-Meeting
  Information</span>](#Pre-Meeting_Information)
  - [<span class="tocnumber">1.1</span> <span class="toctext">Possible
    topics</span>](#Possible_topics)
  - [<span class="tocnumber">1.2</span>
    <span class="toctext">Registration</span>](#Registration)
  - [<span class="tocnumber">1.3</span>
    <span class="toctext">Location</span>](#Location)
- [<span class="tocnumber">2</span>
  <span class="toctext">Attendees</span>](#Attendees)
- [<span class="tocnumber">3</span>
  <span class="toctext">Agenda</span>](#Agenda)
  - [<span class="tocnumber">3.1</span> <span class="toctext">Discussion
    Topics</span>](#Discussion_Topics)
  - [<span class="tocnumber">3.2</span> <span class="toctext">November
    5</span>](#November_5)
  - [<span class="tocnumber">3.3</span> <span class="toctext">November
    6</span>](#November_6)
  - [<span class="tocnumber">3.4</span> <span class="toctext">November
    7</span>](#November_7)
- [<span class="tocnumber">4</span>
  <span class="toctext">Presentations</span>](#Presentations)
- [<span class="tocnumber">5</span> <span class="toctext">Meeting
  Minutes</span>](#Meeting_Minutes)
  - [<span class="tocnumber">5.1</span> <span class="toctext">Big
    Picture</span>](#Big_Picture)
    - [<span class="tocnumber">5.1.1</span> <span class="toctext">GMOD's
      Role</span>](#GMOD.27s_Role)
    - [<span class="tocnumber">5.1.2</span>
      <span class="toctext">Releases and
      Bundles</span>](#Releases_and_Bundles)
    - [<span class="tocnumber">5.1.3</span>
      <span class="toctext">Comparative
      Genomics</span>](#Comparative_Genomics)
  - [<span class="tocnumber">5.2</span> <span class="toctext">GMOD
    Components / Functions</span>](#GMOD_Components_.2F_Functions)
    - [<span class="tocnumber">5.2.1</span>
      <span class="toctext">Apollo</span>](#Apollo)
      - [<span class="tocnumber">5.2.1.1</span>
        <span class="toctext">New Development</span>](#New_Development)
      - [<span class="tocnumber">5.2.1.2</span> <span class="toctext">ID
        Generation and JDBC
        Drivers</span>](#ID_Generation_and_JDBC_Drivers)
    - [<span class="tocnumber">5.2.2</span>
      <span class="toctext">BioPerl, GFF</span>](#BioPerl.2C_GFF)
    - [<span class="tocnumber">5.2.3</span>
      <span class="toctext">Chado</span>](#Chado)
      - [<span class="tocnumber">5.2.3.1</span>
        <span class="toctext">Chado
        Documentation</span>](#Chado_Documentation)
      - [<span class="tocnumber">5.2.3.2</span>
        <span class="toctext">Chado Validator</span>](#Chado_Validator)
      - [<span class="tocnumber">5.2.3.3</span>
        <span class="toctext">DBMS Choice</span>](#DBMS_Choice)
      - [<span class="tocnumber">5.2.3.4</span>
        <span class="toctext">Postgres
        Performance</span>](#Postgres_Performance)
    - [<span class="tocnumber">5.2.4</span>
      <span class="toctext">CMap</span>](#CMap)
    - [<span class="tocnumber">5.2.5</span>
      <span class="toctext">Community
      Annotation</span>](#Community_Annotation)
      - [<span class="tocnumber">5.2.5.1</span>
        <span class="toctext">Community Annotation at
        ParameciumDB</span>](#Community_Annotation_at_ParameciumDB)
      - [<span class="tocnumber">5.2.5.2</span>
        <span class="toctext">Community Annotation at
        JGI</span>](#Community_Annotation_at_JGI)
      - [<span class="tocnumber">5.2.5.3</span>
        <span class="toctext">Community Annotation at
        SGN</span>](#Community_Annotation_at_SGN)
    - [<span class="tocnumber">5.2.6</span>
      <span class="toctext">Community Annotation Server
      (CAS)</span>](#Community_Annotation_Server_.28CAS.29)
    - [<span class="tocnumber">5.2.7</span>
      <span class="toctext">Distributed Annotation System/2
      (DAS/2)</span>](#Distributed_Annotation_System.2F2_.28DAS.2F2.29)
    - [<span class="tocnumber">5.2.8</span>
      <span class="toctext">GBrowse</span>](#GBrowse)
      - [<span class="tocnumber">5.2.8.1</span>
        <span class="toctext">Roadmap</span>](#Roadmap)
      - [<span class="tocnumber">5.2.8.2</span>
        <span class="toctext">Performance</span>](#Performance)
    - [<span class="tocnumber">5.2.9</span> <span class="toctext">Genome
      Grid</span>](#Genome_Grid)
    - [<span class="tocnumber">5.2.10</span> <span class="toctext">Help
      Desk</span>](#Help_Desk)
    - [<span class="tocnumber">5.2.11</span>
      <span class="toctext">Pathway Tools</span>](#Pathway_Tools)
    - [<span class="tocnumber">5.2.12</span>
      <span class="toctext">SynView</span>](#SynView)
    - [<span class="tocnumber">5.2.13</span>
      <span class="toctext">TableEdit</span>](#TableEdit)
    - [<span class="tocnumber">5.2.14</span>
      <span class="toctext">Turnkey, GMODweb,
      DrupalFly</span>](#Turnkey.2C_GMODweb.2C_DrupalFly)
  - [<span class="tocnumber">5.3</span> <span class="toctext">GMOD
    Participating
    Organizations</span>](#GMOD_Participating_Organizations)
    - [<span class="tocnumber">5.3.1</span>
      <span class="toctext">ApiDB</span>](#ApiDB)
      - [<span class="tocnumber">5.3.1.1</span>
        <span class="toctext">Synteny at
        ApiDB</span>](#Synteny_at_ApiDB)
    - [<span class="tocnumber">5.3.2</span>
      <span class="toctext">Berkeley National
      Labs</span>](#Berkeley_National_Labs)
    - [<span class="tocnumber">5.3.3</span>
      <span class="toctext">FlyBase</span>](#FlyBase)
      - [<span class="tocnumber">5.3.3.1</span>
        <span class="toctext">Synteny at
        FlyBase</span>](#Synteny_at_FlyBase)
    - [<span class="tocnumber">5.3.4</span>
      <span class="toctext">GeneDB, Sanger</span>](#GeneDB.2C_Sanger)
    - [<span class="tocnumber">5.3.5</span>
      <span class="toctext">Imperial College
      London</span>](#Imperial_College_London)
    - [<span class="tocnumber">5.3.6</span> <span class="toctext">JCVI
      (nee TIGR)</span>](#JCVI_.28nee_TIGR.29)
    - [<span class="tocnumber">5.3.7</span>
      <span class="toctext">MaizeGDB</span>](#MaizeGDB)
    - [<span class="tocnumber">5.3.8</span>
      <span class="toctext">ParameciumDB</span>](#ParameciumDB)
    - [<span class="tocnumber">5.3.9</span>
      <span class="toctext">Riken</span>](#Riken)
    - [<span class="tocnumber">5.3.10</span>
      <span class="toctext">University of Maryland Medical
      Center</span>](#University_of_Maryland_Medical_Center)
    - [<span class="tocnumber">5.3.11</span>
      <span class="toctext">WormBase / CSHL</span>](#WormBase_.2F_CSHL)
      - [<span class="tocnumber">5.3.11.1</span>
        <span class="toctext">GBrowse_Syn</span>](#GBrowse_Syn)


## <span id="Pre-Meeting_Information" class="mw-headline">Pre-Meeting Information</span>

### <span id="Possible_topics" class="mw-headline">Possible topics</span>

A list of suggested topics, raised in advance by GMOD community members.

- community annotation - FlyBase seconds this topic
- Chado standard on ortholog/paralog/synteny storage.
- The state of [GFF](GFF "GFF") tools in [BioPerl](BioPerl "BioPerl").
  Some of the auditing and examples are on a
  <a href="http://bioperl.org/wiki/GFF_code_audit" class="external text"
  rel="nofollow">Bioperl wiki page</a>.
- GMOD releases and packaging
  - How hard would it be to heap together specific releases of popular
    GMOD components into a named/numbered release that has gone through
    some level of compatibility testing?
  - How much pain does a lack of such a release currently cause users?
  - how much might the community annotation server help with this?

### <span id="Registration" class="mw-headline">Registration</span>

There was a \$25 registration fee to cover meals and other costs
associated with the meeting. Please contact Scott Cain
<a href="mailto:cain@cshl.edu" class="external text"
rel="nofollow">cain@cshl.edu</a> if you need a reciept for your payment.

### <span id="Location" class="mw-headline">Location</span>

The meeting was held at
<a href="http://www.cshl.edu/" class="external text" rel="nofollow">Cold
Spring Harbor Laboratory's</a> at the <a
href="http://maps.google.com/maps?f=q&amp;hl=en&amp;geocode=&amp;time=&amp;date=&amp;ttype=&amp;q=500+Sunnyside+Boulevard,+Woodbury+NY&amp;sll=40.800589,-73.468398&amp;sspn=0.007407,0.009849&amp;ie=UTF8&amp;ll=40.800556,-73.468881&amp;spn=0.007407,0.009849&amp;t=h&amp;z=16&amp;iwloc=addr&amp;om=1"
class="external text" rel="nofollow">Woodbury building</a>, which is not
on the main CSHL campus.

  

## <span id="Attendees" class="mw-headline">Attendees</span>

- James Abbott, Imperial College, London
- Sam Angiuoli, University of Maryland Medical School
- Tim Burgis, Imperial College, London
- [Scott Cain](User%3AScott "User%3AScott"), GMOD Coordinator
- [Mike Caudy](User%3AMcaudy "User%3AMcaudy"), CSHL
- [Dave Clements](User%3AClements "User%3AClements"), [GMOD Help
  Desk](GMOD_Help_Desk "GMOD Help Desk"),
  <a href="http://nescent.org" class="external text"
  rel="nofollow">NESCent</a>
- Norie de la Cruz, WormBase
- Quenfen Dong, Indiana University
- Dave Emmert, <a href="http://flybase.org" class="external text"
  rel="nofollow">FlyBase</a>
- [Ben Faga](User%3AFaga "User%3AFaga"), CSHL
- Kathleen Falls, <a href="http://flybase.org" class="external text"
  rel="nofollow">FlyBase</a>
- [Steve Fischer](User%3AStevef "User%3AStevef"),
  <a href="http://apidb.org" class="external text"
  rel="nofollow">ApiDB</a>
- [Don Gilbert](User%3ADongilbert "User%3ADongilbert")
- [Josh Goodman](User%3AJogoodma "User%3AJogoodma"), FlyBase - Indiana
  University
- [Jay Hannah](User%3AJhannah "User%3AJhannah"), University of Nebraska
- [Todd Harris](User%3ATharris "User%3ATharris"), WormBase - Cold Spring
  Harbor Laboratory
- Sven Heinicke, Princeton
- Kevin Galens, JCVI
- [Gregg Helt](User%3AGreggHelt2 "User%3AGreggHelt2"), DAS/2
- [Chris Hemmerich](User%3AChemmeri "User%3AChemmeri"),
  <a href="http://flybase.org" class="external text"
  rel="nofollow">FlyBase</a>
- Hideya Kiwaji, <a href="http://www.riken.go.jp/" class="external text"
  rel="nofollow">Riken</a>
- [Ed Lee](User%3AElee "User%3AElee"), Lawrence Berkeley Labs
- Suzi Lewis, <a href="http://bioontology.org/" class="external text"
  rel="nofollow">National Center for Biomedical Ontology</a>
- [Sheldon McKay](User%3AMckays "User%3AMckays"), WormBase/modENCODE - Cold
  Spring Harbor Laboratory
- Lukas Mueller,
  <a href="http://soldb.cit.cornell.edu/" class="external text"
  rel="nofollow">Sol Genomics Network</a>
- [Joshua Orvis](User%3AJorvis "User%3AJorvis"), University of Maryland
  Medical Center
- Suzanne Paley, <a href="http://ecocyc.org" class="external text"
  rel="nofollow">EcoCyc</a>
- Chinmay Patel, GeneDB, Sanger Institute
- David Riley, University of Maryland Medical Center
- Andy Schroeder, <a href="http://flybase.org" class="external text"
  rel="nofollow">FlyBase</a>
- Taner Sen, <a href="http://maizegdb.org" class="external text"
  rel="nofollow">MaizeGDB</a>
- [Linda Sperling](User%3ASperling "User%3ASperling"),
  [ParameciumDB](ParameciumDB "ParameciumDB") - CNRS
- [Jason Stajich](User%3AStajich "User%3AStajich")
- [Lincoln Stein](User%3ALstein "User%3ALstein"), CSHL
- Victor Strelets, <a href="http://flybase.org" class="external text"
  rel="nofollow">FlyBase</a>
- Haiming Wang <a href="http://apidb.org" class="external text"
  rel="nofollow">ApiDB.org</a>
- Robert Wilson, <a href="http://flybase.org" class="external text"
  rel="nofollow">FlyBase</a>
- Haiyan Zhang, <a href="http://flybase.org" class="external text"
  rel="nofollow">FlyBase</a>
- Pinglei Zhou, <a href="http://flybase.org" class="external text"
  rel="nofollow">FlyBase</a>

## <span id="Agenda" class="mw-headline">Agenda</span>

### <span id="Discussion_Topics" class="mw-headline">Discussion Topics</span>

We spent some time on our first day discussion what topics attendees
would like to discusss. This list of topics helped shape the meeting
agenda.

- Community Annotation
  - DAS, Apollo, genome-Wiki
- Comparative Genomics
  - Synteny viewers
  - Chado data storage
  - See [Chado Comparative
    Schema](Chado_Comparative_Schema "Chado Comparative Schema").
- BioPerl and [GFF(2/3)](GFF "GFF")
  - [GFF](GFF "GFF") Questions

<!-- -->

- Postgres Tuning / [Materialized
  views](Materialized_views "Materialized views")
  - Performance Strategies
- Apollo-Chado Connection
  - Performance - See [PostgreSQL Performance
    Tips](PostgreSQL_Performance_Tips "PostgreSQL Performance Tips").
  - Too many JDBC Adaptors
- Chado
  - ID Generation
  - Moving away from Postgres
  - Missing Chado pieces (phylogenetics)

<!-- -->

- What Should GMOD Focus On (What's Missing)
  - Genome Analysis ([Galaxy](Galaxy.1 "Galaxy"),
    <a href="Ergatis" class="mw-redirect" title="Ergatis">Ergatis</a>,
    ...)
    - Lightweight annotation
      <a href="http://www.yandell-lab.org/maker/index.html"
      class="external text" rel="nofollow">MAKER pipeline</a> from Mark
      Yandell  
      (*2008/05/13: [MAKER](MAKER.1 "MAKER") has since been folded in to
      GMOD.*)
  - MicroArrays
  - What is the GMOD Community and how best can we serve them?
  - Is there a need for individual MODs?
- What should [GMOD Help Desk](GMOD_Help_Desk "GMOD Help Desk") do?
  - UIs: Picture Intensive
- What should be the outcome of this meeting?

### <span id="November_5" class="mw-headline">November 5</span>

1:00 Shuttle from Grace Auditorium to Woodbury

1:30 Introductions

2:30 Coffee

5:30 Shuttle from Woodbury to Grace Auditorium

### <span id="November_6" class="mw-headline">November 6</span>

8:50 Shuttle from Grace Auditorium to Woodbury

9:15Â ? Scott

10:15 Community Annotation

- Linda Sperling - ParameciumDB
- Lukas Muller - SGN

10:30 Coffee

- Michael Caudy - FlyBase Drupal

12:00 Lunch

1:00 Standards and applications for storing comparative genome data

- [Steve Fisher](User%3AStevef "User%3AStevef") -
  [GBrowse](GBrowse.1 "GBrowse"): [SynView](SynView "SynView") and the
  Generic database adaptor
- Victor Strelets - FlyBase Orthoview (GBrowse)

2:30 Coffee

- Sheldon McKay - gbrowse_syn

  

5:30 Shuttle from Woodbury to Grace Auditorium

### <span id="November_7" class="mw-headline">November 7</span>

8:50 Shuttle from Grace Auditorium to Woodbury

9:15 BioPerl

- [GFF3](GFF3 "GFF3") tools
- SeqFeatures/FeatureIO
- Sequence Ontology

10:30 Coffee

12:00 Shuttle from Woodbury to Grace Auditorium

  

## <span id="Presentations" class="mw-headline">Presentations</span>

- <a href="http://eugenes.org/gmod/docs/gmod-update-07nov.ppt"
  class="external text" rel="nofollow">GMOD Indiana update</a> slides,
  Don Gilbert
- WormBase update, [Todd Harris](User%3ATharris "User%3ATharris"); Slides:
  <a
  href="http://dev.wormbase.org/presentations/2007/2007.11-GMOD-WormBase/2007.11-GMOD-WormBase.key.tgz"
  class="external text" rel="nofollow">Keynote</a>, <a
  href="http://dev.wormbase.org/presentations/2007/2007.11-GMOD-WormBase/2007.11-GMOD-WormBase.ppt"
  class="external text" rel="nofollow">Powerpoint</a>, <a
  href="http://dev.wormbase.org/presentations/2007/2007.11-GMOD-WormBase/2007.11-GMOD-WormBase.pdf"
  class="external text" rel="nofollow">PDF</a>, <a
  href="http://dev.wormbase.org/presentations/2007/2007.11-GMOD-WormBase/2007.11-GMOD-WormBase.mov"
  class="external text" rel="nofollow">Mov</a>
- <a
  href="http://mango.ctegd.uga.edu/jkissingLab/presentations/GMOD_Nov_2007.ppt"
  class="external text" rel="nofollow">ApiDB GBrowse update</a> slides,
  Haiming Wang
- <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/3/38/2007_11_05_CMap_GMOD.ppt"
  class="internal" title="2007 11 05 CMap GMOD.ppt">CMap/CMAE Progress
  Report</a>, [Ben Faga](User%3AFaga "User%3AFaga")
- <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/5/58/Gbrowse_syn.pdf" class="internal"
  title="Gbrowse syn.pdf">Gbrowse_syn</a> Sheldon McKay
- <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/a/ac/CommunityAnnotationNov2007.pdf"
  class="internal" title="CommunityAnnotationNov2007.pdf">Community
  Annotation</a> [Linda Sperling](User%3ASperling "User%3ASperling")
- <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/b/be/Workshop.pdf" class="internal"
  title="Workshop.pdf">Community Annotation</a> Chinmay Patel
- <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/1/19/SyntenyModeling.pdf" class="internal"
  title="SyntenyModeling.pdf">Modeling and Displaying Synteny w/
  SynView</a> [Steve Fischer](User%3AStevef "User%3AStevef")
- <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/c/c8/GMOD-Nov-2007.ppt" class="internal"
  title="GMOD-Nov-2007.ppt">Recent Developments in Pathway Tools</a>,
  Suzanne Paley

## <span id="Meeting_Minutes" class="mw-headline">Meeting Minutes</span>

The minutes here are based on Dave Clements' notes from the meeting.
They are far from complete and you are encouraged to expand and correct
them.

The minutes are not chronological. Rather they are broken up into 3
sections:

- [Big Picture](#Big_Picture)
- [GMOD Components / Functions](#GMOD_Components_.2F_Functions)
- [GMOD Participating Organizations](#GMOD_Participating_Organizations)

### <span id="Big_Picture" class="mw-headline">Big Picture</span>

We had several discussions about *the big picture.*

  

#### <span id="GMOD.27s_Role" class="mw-headline">GMOD's Role</span>

Don Gilbert pointed out that cheap short sequencers are now available.
Lots of people have inexpensive sequnces, but there still is no way to
do cheap annotation.

Current GMOD clients are species or family centered. Want to make it
easy to integrate multiple species. ApiDB is at the point of opening new
species databases and web sites with relatively little effort.

Comparative genomics came up over and over again, both across species
and within species.

As data grows and is consolidated, issues of who owns the data and who's
responsible for the annotation become more problematic.

How does GMOD want to deal with integration issues?

How close to the sequencer does GMOD want to get? We don't want to pull
the data off the sequencer.

Should we position GMOD as something that can feed data into places like
Ensembl? Ensembl does not have curation expertise of the MODs. Even if
NCBI is wonderful at consolidation, they won't have quality curation.
GMOD sits right there, supporting curation. So, we doubt that Ensembl or
NCBI will swallow us whole.

  

#### <span id="Releases_and_Bundles" class="mw-headline">Releases and Bundles</span>

We need to figure out what components we want and what we are pushing.
If we focus on a core set of packages then life gets easier for the
project.

There was discussion of better release management for components, and
the VMWare Community Annotation Server package. Are GMOD bundles the way
of the future? Believe that binary packages are generally not going to
work for GMOD unless someone is willing to put a lot of time into
maintaining them.

  

#### <span id="Comparative_Genomics" class="mw-headline">Comparative Genomics</span>

Comparative genomics came up over and over again, both across species
and within species. The GBrowse_syn talk in particular spawned a
discussion on this.

First, can Chado represent relationships that have more than two
members? Yes. Feature_loc has a rank column. Do we want collections in
Chado?

Jason suggested a working group on how to do this. Dave from UMD
volunteered to manage a wiki page on this, with the end goal of
establishing a document that defines how to store comparative genomes.

Talks on synteny are spread throughout this document.

  

### <span id="GMOD_Components_.2F_Functions" class="mw-headline">GMOD Components / Functions</span>

#### <span id="Apollo" class="mw-headline">Apollo</span>

##### <span id="New_Development" class="mw-headline">New Development</span>

Work has resumed on developing [Apollo](Apollo.1 "Apollo"). [Ed
Lee](User%3AElee "User%3AElee") formerly of TIGR/JCVI started working for
Suzi Lewis at Berkeley this fall and is working on it. Work is being
done on

- A [GFF3](GFF3 "GFF3") adapter
- Speeding up Apollo when it uses
  <a href="Chado" class="mw-redirect" title="Chado">Chado</a> as a
  backend (or, just speeding up Chado).
- Communicating with more than one Chado instance.
- Undo/Redo support.

##### <span id="ID_Generation_and_JDBC_Drivers" class="mw-headline">ID Generation and JDBC Drivers</span>

Apollo can talk directly to a database or it can use XML files instead.
FlyBase, VectorBase, BeeBase, and BovineBase are all believed to take
the XML approach.

Apollo currently has two choices for database adaptors:

1.  One that uses Postgres database triggers to set IDs.
2.  One that does not.

The trigger version is used in the Community Annotation Server and on
the Dolan-Rice project. We could not think of anywhere else it was used.
The triggerless version is used everywhere else that we knew of.

The trigger version is Postgres specific. The triggerless version stores
multiple copies of shared exons.

Notes from Tuesday: Decided to actively discourage use of the trigger
version. Best thing may be to go through trigger code and externalize
the logic.

Notes from Wednesday: Apollo - Chado - No short term decision. Long term
probably move to Crabtree.

As you may have noticed, those notes disagree.

  

#### <span id="BioPerl.2C_GFF" class="mw-headline">BioPerl, GFF</span>

There was a discussion of [BioPerl](BioPerl "BioPerl") and how it
relates to GMOD.

Jason Stajich created a slimmed down feature Perl package based on
arrays instead of hashes: Bio::SeqFeature::Slim. This is 70% faster for
reading a [GFF](GFF "GFF") file. Bio::Feature::IO only supports
[GFF3](GFF3 "GFF3"). It is slow, uses heavy objects, and is strongly
typed. Jason wants to spend more time on middleware speed. He also wants
converter into a common object model and code to get it back out to any
supported format.

6 to 8 people are currently contributing to BioPerl.

  
[GFF3](GFF3 "GFF3") has an ID field. ID is not clear in earlier
versions. [GFF2](GFF2 "GFF2") supports arbitrary feature types. GFF3
requires SO types (but you can always ignore that). Keep detailed
alignment data in a separate database, not in GFF3. Indicate in GFF3
that data is stored elsewhere. Could store cigar strings in GFF3 and
spec supports that.

  

  

#### <span id="Chado" class="mw-headline">Chado</span>

There was a request to make to Chado be more database neutral, rather
than Postgres-specific.

The slowness of Chado databases came up in several contexts. David from
UMD Medical Center started a Postgres performance page on the wiki.

Scott described a potential way to implement materialized views in Chado
that gets us most of the benefits of DBMS-supported materialized views.
Store

- the SQL to create it in a table,
- a run time schedule for when the table should be rebuilt,
- an enabled/disabled flag that is disabled by default.

Question was raised if genome metadata fits into the current Chado. The
belief was that it does not.

Jason Stajich wants a better idea of who is responsible for what in
terms of Chado modules. Dave C will take this on.

  

##### <span id="Chado_Documentation" class="mw-headline">Chado Documentation</span>

The table level and column level documentation for Chado is in a good
state. Enhanced basic, big picture documentation was requested. Josh
Goodman is thinking of providing a mapping from Chado DB columns to
FlyBase report columns. Mike Caudy pointed out we should have multiple
examples of implementation, not just FlyBase.

  

##### <span id="Chado_Validator" class="mw-headline">Chado Validator</span>

We discussed if a
<a href="Chado" class="mw-redirect" title="Chado">Chado</a> database
validator would be worthwhile. A validator would check a Chado database
to see if it conforms to the canonical model for a Chado database. There
was no consensus on the value or practicality of this. There was
consensus that no one was willing to volunteer to write it.

Ben suggested that if and when we do this, we use the
[GFF3](GFF3 "GFF3") to Chado validator as a starting point.

  

##### <span id="DBMS_Choice" class="mw-headline">DBMS Choice</span>

There was a request to make to Chado be more database neutral, rather
than Postgres-specific. Someone also asked if there was an SQLite
adapter for GBrowse.

  

##### <span id="Postgres_Performance" class="mw-headline">Postgres Performance</span>

Slow performance of Chado Postgres implementations came up repeatedly.

Some bits:

- Specify locale. ASCII-US runs fast. UTF-8 is slow and that is the
  default. Specified for the server, at server start.
- A lot of time has been spent on making the queries go fast.
- RTree indexes are in the core.
- Allen's FRange functions are in the DB, but aren't used by default
  queries.

  

#### <span id="CMap" class="mw-headline">CMap</span>

**Presentation:**
<a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/3/38/2007_11_05_CMap_GMOD.ppt"
class="internal" title="2007 11 05 CMap GMOD.ppt">CMap Progress
Report</a>, [Ben Faga](User%3AFaga "User%3AFaga")

New CMap release (1.0) is on its way. Will have an assembly editor.
Includes a dot plot, new glyphs, and an install script based on the
GBrowse install script.

Ben will ask users to do beta testing, and hopes to start with that
before end of 2007. Ben is looking for a project that is doing large
scale assembly, to test CMap for doing assembly correction.

#### <span id="Community_Annotation" class="mw-headline">Community Annotation</span>

This was a popular motif in the meeting.

  

##### <span id="Community_Annotation_at_ParameciumDB" class="mw-headline">Community Annotation at ParameciumDB</span>

**Presentation:**
<a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/a/ac/CommunityAnnotationNov2007.pdf"
class="internal" title="CommunityAnnotationNov2007.pdf">Community
Annotation</a>, [Linda Sperling](User%3ASperling "User%3ASperling")

Linda Sperling discussed ParameciumDB. Paramecium is a small community
with few resources and no dedicated curators.

Paramecium curators are a small set of people that must do their
annotation from fixed IP addresses. Curator annotations are kept in
addition to existing Genoscope predictions. These annotation are not
validated when they are submitted. Annotators cannot chage annotations
made by other people. There are two databases: one backing the website,
and one where annotation goes. Once a month the new annotation is pushed
to the web site. Validation happens prior to release.

They are also using ParameciumDB to teach annotation at two colleges,
and some annotation comes from that. The bulk of annotations come from 2
curators, with the other curators all making a small number of
annotations.

Uses Java WebStart version of Apollo. Annotators click on link and
Apollo starts up. Apollo talks directly to Chado, using the triggerless
database adapter.

##### <span id="Community_Annotation_at_JGI" class="mw-headline">Community Annotation at JGI</span>

Don Gilbert briefly described community annotation at JGI. They have a
web interface for simple annotations and use Apollo for complex
annotations. Anyone can promote any gene model, but they can't delete
other models. Use the Wikipedia model: Whoever annotates last is
correct.

  

##### <span id="Community_Annotation_at_SGN" class="mw-headline">Community Annotation at SGN</span>

Lukas Mueller discussed [SGN](Category%3ASGN "Category%3ASGN").

SGN has data for tomato, potato, eggplant, and many other species. SGN
is locus centric. Each locus has (or can have) a single person who is
the editor/owner of that locus. The locus editor can change anything
about that locus that they want. The name of the locus editor is
displayed on the locus page. Every locus has a "request editor
privileges" link, if that locus has been assigned or not.

All edits are logged, and nothing is ever truly deleted. 'Deleted' items
are retained but flagged as obsolete and are no longer shown.

SGN supports tagging of loci. Tags are free text that are rationalized
after they are created. The tagging metaphor for curation also came up
in several contexts during the Genome Informatics meeting.

#### <span id="Community_Annotation_Server_.28CAS.29" class="mw-headline">Community Annotation Server (CAS)</span>

Scott Cain spoke about this. It is almost ready to go. The Community
Annotation Server (CAS) is meant to be "GMOD in a box". Currently it
consists of:

- A VMWare image, containing
- Ubuntu Linux, version 6.10 LTS.
  - Picked Ubuntu LTS over CentOS because LTS stands for *long term
    service* and it will be supported for a while.
- Postgres
- A <a href="Chado" class="mw-redirect" title="Chado">Chado</a> database
  with DictyBase data in it.
- An empty Chado database
- [Modware](Modware "Modware")
- [Apollo](Apollo.1 "Apollo") - Uses the JDBC adaptor with triggers.
  This is a Java WebStart version.
- [GBrowse](GBrowse.1 "GBrowse")
- MediaWiki - includes Cite, ProcessCite and TableEdit extensions.
  - Cite extensions make it easy to provide literature annotations.
    Provide PubMed ID and it finds and grabs extract from PubMed.

Note that it does not include Turnkey and/or
[GMODWeb](GMODWeb "GMODWeb"). [Lincoln](User%3ALstein "User%3ALstein") would
like to add GMODweb, [Textpresso](Textpresso "Textpresso") and
[BioMart](BioMart "BioMart") to that list.

This can run on any Intel machine, including Apple. Very little
performance hit is caused by virtualization.

An online trial version of the Community Annotation Server was requested
and was already on the way.

  

#### <span id="Distributed_Annotation_System.2F2_.28DAS.2F2.29" class="mw-headline">Distributed Annotation System/2 (DAS/2)</span>

Gregg Helt attended with the goal of bringing the [Distributed
Annotation
System](Distributed_Annotation_System "Distributed Annotation System"),
version 2 (DAS/2) into the GMOD family.

Preserving DAS/1 Strengths in DAS/2

- Keep focus on location-based annotation of biological sequences.
- Protocol, not an implementation.
  - HTTP for transport,
  - URLs for queries
  - XML for responses
  - REST-like style.
- No Required central authority.
- Couple XML response to URL request formats.
- XML has been shortened, but big gain comes from client-server content
  format negotiation, including binary. Empty elements dropped.
- Uses HTTP caching in the client.
- [IGB](IGB "IGB") - reference client for DAS2. Integrated Genome
  Browser

Allen Day built a DAS2 server on top of
<a href="Chado" class="mw-redirect" title="Chado">Chado</a>. That is in
CVS.

There is a validation suite for server responses to different queries.

Spec has not changed in over a year.

Scott would like that when someone installs Chado, they also get
[BioMart](BioMart "BioMart") and DAS2. That is, they get access by
default. Gregg would like to see GBrowse get a DAS/2 adapter.

#### <span id="GBrowse" class="mw-headline">GBrowse</span>

##### <span id="Roadmap" class="mw-headline">Roadmap</span>

[Lincoln Stein](User%3ALstein "User%3ALstein") talked about upcoming
releases of [GBrowse](GBrowse.1 "GBrowse").

- 1.69
  - Is in pre-release state.
  - Has
    - popups
    - drag tracks vertically
    - quantitative data
    - multiple alignment and conservation tracks.
- 1.7
  - Release by end of year
  - Rubberbanding (zoom by selecting a rectangle with mouse)
  - Autocomplete
- 2.0
  - Release in early 2008
  - Major performance and scalability enhancements.
    - e.g., each track can be drawn by different server or CPU.
- 3.0 *(subsequently renamed to [JBrowse](JBrowse.1 "JBrowse"))*
  - Released sometime in 2008
  - Google maps type interface.
    - e.g., zooming and panning via mouse.

Version 3.0 (now called [JBrowse](JBrowse.1 "JBrowse")) is a fork of the
code and version 2 and 3 are expected to co-exist 'forever'. Some shops
won't have the horsepower to power version 3, and Lincoln wants to keep
it as an easy to install tool.

##### <span id="Performance" class="mw-headline">Performance</span>

<a href="Chado" class="mw-redirect" title="Chado">Chado</a> is usually
too slow to run [GBrowse](GBrowse.1 "GBrowse") on top of. Consider using
Bio::DB:GFF instead. (Can't run GBrowse on top of
[BioMart](BioMart "BioMart"). No adapter exists because of BioMart's
flexible schema.)

Jason S argues that GBrowse slows down when it does BioPerl object
creation. These are relatively heavyweight objects. He has just written
a Slim version that is up to 70% faster.

Browser speed was also the number one issue (with all browsers) at the
Genome Browsers Birds-of-a-Feather meeting at Genome Informatics.

  

#### <span id="Genome_Grid" class="mw-headline">Genome Grid</span>

**Presentation:**
<a href="http://eugenes.org/gmod/docs/gmod-update-07nov.ppt"
class="external text" rel="nofollow">GMOD Indiana update</a> slides, Don
Gilbert

Don Gilbert spoke about [Genome grid](Genome_grid "Genome grid").

Genome Grid is middleware to enable easy use of TeraGrid for genome
analysis tasks. Don is looking for genomes that need compute intensive
analysis. He also interested in applying [BioMart](BioMart "BioMart")
and <a href="Ergatis" class="mw-redirect" title="Ergatis">Ergatis</a> to
these problems.

#### <span id="Help_Desk" class="mw-headline">Help Desk</span>

Dave Clements introduced himself and the goals of the [GMOD Help
Desk](GMOD_Help_Desk "GMOD Help Desk") position.

Dave will make the help desk more visible on the web site, and add a
[GMOD News](GMOD_News "GMOD News") column to the [home
page](Main_Page "Main Page").

#### <span id="Pathway_Tools" class="mw-headline">Pathway Tools</span>

**Presentation:**
<a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/c/c8/GMOD-Nov-2007.ppt" class="internal"
title="GMOD-Nov-2007.ppt">Recent Developments in Pathway Tools</a>

Suzanne Paley talked about recent developments in [Pathway
Tools](Pathway_Tools.1 "Pathway Tools"), including:

- Advanced Query Form
- Richer representation of regulation
- Pathlogic over-infers pathways. Pathways now have to be tagged to be
  shown.
- Dataset diffs and incremental updates.

#### <span id="SynView" class="mw-headline">SynView</span>

**Presentation:**
<a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/1/19/SyntenyModeling.pdf" class="internal"
title="SyntenyModeling.pdf">Modeling and Displaying Synteny w/
SynView</a>, [Steve Fischer](User%3AStevef "User%3AStevef")

[Steve Fischer](User%3AStevef "User%3AStevef") of ApiDB (see below) spoke
about SynView. SynView is a synteny browser based on GBrowse. It is
described in a <a
href="http://bioinformatics.oxfordjournals.org/cgi/content/full/22/18/2308"
class="external text" rel="nofollow">Bioinformatics paper</a>.

His talked raised a number of issues that have come up with recent
extensions to SynView.

#### <span id="TableEdit" class="mw-headline">TableEdit</span>

This is a MediaWiki extension by Jim Hu. It does two things. First, it
makes it easier to update tables in MediaWiki, by presenting a nicer
interface for altering wiki tables. Secondly, it supports synchronizing
MediaWiki tables from database tables and vice versa.

  

  

#### <span id="Turnkey.2C_GMODweb.2C_DrupalFly" class="mw-headline">Turnkey, GMODweb, DrupalFly</span>

These are all web interface layers that lay on top of Chado databases.

[GMODWeb](GMODWeb "GMODWeb") is currently not working, we think because
SQLTranslator has not been upgraded to deal with recent versions of
Postgres. [Ben Faga](User%3AFaga "User%3AFaga") agreed to actively work on
this.

Michael Caudy argued that even if GMODWeb did work right now that it is
not extensible enough to support complex queries and presentation. Mike
presented Drupal, Drupal Views, and PHPTemplate as an alternative web
framework for providing a web interface to Chado databases. Mike
demonstrated a prototype called DrupalFly that presents FlyBase data in
an alternative organization.

[Lincoln](User%3ALstein "User%3ALstein") has an opening in Toronto for a
full time programmer. Lincoln will talk with Brian about
[GMODWeb](GMODWeb "GMODWeb")'s future. We will put something on web site
asking for volunteers to take on GMODweb.

  

  

### <span id="GMOD_Participating_Organizations" class="mw-headline">GMOD Participating Organizations</span>

A number of organizations talked about their recent work.

  

#### <span id="ApiDB" class="mw-headline">ApiDB</span>

**Presentation:** <a
href="http://mango.ctegd.uga.edu/jkissingLab/presentations/GMOD_Nov_2007.ppt"
class="external text" rel="nofollow">ApiDB GBrowse update</a>, Haiming
Wang

[Steve Fischer](User%3AStevef "User%3AStevef") talked about ApiDB. ApiDB
uses GUS as their schema. They do multispecies comparative analysis.
They have a database adapter link from GBrowse to GUS. It is based on
the Chado adapter. They use materialized views in Oracle 10G and it is
still relatively slow.

  

##### <span id="Synteny_at_ApiDB" class="mw-headline">Synteny at ApiDB</span>

See [SynView](#Synview) above for details on SynView.

Syntenic maps at ApiDB are produced with Mercator. The maps are based on
gene orthology. Gene orthologs are generated using OrthoMCL. All
alignments are pairwise, rather than multiple. Orthology is represented
outside standard GUS schema. In the synteny schema, everything is
defined relative to the reference sequence. Also need a table to define
anchors.

[Steve Fischer](User%3AStevef "User%3AStevef") showed an 11 track page,
which has about 5000 popups in it.

ApiDB has a release cycle. They discard and recalculate synteny with
every new release.

  

#### <span id="Berkeley_National_Labs" class="mw-headline">Berkeley National Labs</span>

The Berkeley group is actively involved in supporting and developing
<a href="Chado" class="mw-redirect" title="Chado">Chado</a>, GO, SO,
OBO-Edit, [Phenote](Phenote "Phenote"), [Apollo](Apollo.1 "Apollo"), and
the new [AJAX](Glossary#AJAX "Glossary") [GBrowse](GBrowse.1 "GBrowse").

#### <span id="FlyBase" class="mw-headline">FlyBase</span>

[FlyBase](Category%3AFlyBase "Category%3AFlyBase") has migrated their
production databases to the
<a href="Chado" class="mw-redirect" title="Chado">Chado</a> [database
schema](Glossary#Database_Schema "Glossary"). FlyBase uses:

- <a href="Chado" class="mw-redirect" title="Chado">Chado</a>
- GMOD [XORT](XORT.1 "XORT")
- [Chado XML](Chado_XML "Chado XML")
- [Apollo](Apollo.1 "Apollo")
- [BioMart](BioMart "BioMart")

  

##### <span id="Synteny_at_FlyBase" class="mw-headline">Synteny at FlyBase</span>

Victor Strelets talked about OrthoView, an extension to
[GBrowse](GBrowse.1 "GBrowse") for viewing
<a href="Synteny" class="mw-redirect" title="Synteny">synteny</a>.

Victor also presented the genetic interactions viewer, a fast way of
visualizing gene interactions. It does not run directly off of the
<a href="Chado" class="mw-redirect" title="Chado">Chado</a> database.

#### <span id="GeneDB.2C_Sanger" class="mw-headline">GeneDB, Sanger</span>

**Presentation:**
<a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/b/be/Workshop.pdf" class="internal"
title="Workshop.pdf">Community Annotation</a>, Chinmay Patel

Chinmay Patel spoke about a week-long annotation project at Sanger
involving 40 people all annotating the same genome.

They used the [Artemis](Artemis "Artemis") annotation editor (instead of
[Apollo](Apollo.1 "Apollo")), but Artemis was talking to a
<a href="Chado" class="mw-redirect" title="Chado">Chado</a> database
using an Artemis-Chado Ibatis-based (instead of Hibernate-based)
adapter. The adapter is not yet released. (*But it is now: see
[Artemis-Chado Integration
Tutorial](Artemis-Chado_Integration_Tutorial "Artemis-Chado Integration Tutorial").*)

#### <span id="Imperial_College_London" class="mw-headline">Imperial College London</span>

Using GMOD to support a fungal sequencing project. Using:

- Chado
- GBrowse
- Apollo

  

#### <span id="JCVI_.28nee_TIGR.29" class="mw-headline">JCVI (nee TIGR)</span>

Using Chado as database schema.

  

#### <span id="MaizeGDB" class="mw-headline">MaizeGDB</span>

Taner Sen from MaizeGDB was at the meeting. Maize has multiple groups
generating different gene models. It would be nice to display each groun
in a separate track. MaizeGDB is evaluating genome browsers and is
considering using GBrowse.

#### <span id="ParameciumDB" class="mw-headline">ParameciumDB</span>

**Presentation:**
<a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/a/ac/CommunityAnnotationNov2007.pdf"
class="internal" title="CommunityAnnotationNov2007.pdf">Community
Annotation</a>, [Linda Sperling](User%3ASperling "User%3ASperling")

Use GMOD for almost everything:

- Chado
- Apollo
- Turnkey
- GBrowse

  
Paramecium is an odd critter (unicellular eukaryote, ciliate clade):

- 72 Mbp
- 40K gene models
  - 12,500 computationally identified potential errors.
- At least 3 whole genome duplication events.
- Nuclear dimorphism. Germline nucleus (not yet sequenced) and somatic
  nucleus (sequenced) which is a rearranged version of the germline
  nucleus, streamlined for gene expression.

Fewer than 20 paramecium molecular biology labs in the world. Database
supported with 1.5 staff.

It is important that people be able to click on a link, launch Apollo,
add some curation and save it. Their Apollo talks directly to Chado (no
triggers). See Community Annotation above for more.

#### <span id="Riken" class="mw-headline">Riken</span>

Riken uses GBrowse.

  

  

#### <span id="University_of_Maryland_Medical_Center" class="mw-headline">University of Maryland Medical Center</span>

Use <a href="Chado" class="mw-redirect" title="Chado">Chado</a> as a
backend, a lot. Use [Sybil](Sybil "Sybil") for [comparative
genomics](Category%3AComparative_Genomics "Category%3AComparative Genomics"),
and are a mix of [PostgreSQL](PostgreSQL "PostgreSQL") and Oracle.

#### <span id="WormBase_.2F_CSHL" class="mw-headline">WormBase / CSHL</span>

**Presentation:** <a
href="http://dev.wormbase.org/presentations/2007/2007.11-GMOD-WormBase/2007.11-GMOD-WormBase.key.tgz"
class="external text" rel="nofollow">Keynote</a>, <a
href="http://dev.wormbase.org/presentations/2007/2007.11-GMOD-WormBase/2007.11-GMOD-WormBase.ppt"
class="external text" rel="nofollow">Powerpoint</a>, <a
href="http://dev.wormbase.org/presentations/2007/2007.11-GMOD-WormBase/2007.11-GMOD-WormBase.pdf"
class="external text" rel="nofollow">PDF</a>, <a
href="http://dev.wormbase.org/presentations/2007/2007.11-GMOD-WormBase/2007.11-GMOD-WormBase.mov"
class="external text" rel="nofollow">Mov</a>, [Todd
Harris](User%3ATharris "User%3ATharris")

Wormbase is migrating to Chado slowly. There is currently very little
Chado there.

  

##### <span id="GBrowse_Syn" class="mw-headline">GBrowse_Syn</span>

**Presentation:**
<a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/5/58/Gbrowse_syn.pdf" class="internal"
title="Gbrowse syn.pdf">Gbrowse_syn</a>, [Sheldon
McKay](User%3AMckays "User%3AMckays")

Sheldon McKay talked about [GBrowse_syn](GBrowse_syn.1 "GBrowse syn"), a
prototype extension to [GBrowse](GBrowse.1 "GBrowse") for viewing
<a href="Synteny" class="mw-redirect" title="Synteny">synteny</a>. Goal
is to have a *sequence* alignment viewer that can look at more than two
species at a time. GBrowse_syn is based purely on sequence alignments.
It does not know about genes or orthologs per se.

  
Used PECAN for the alignments. Maps are precomputed in a very
CPU-intensive step.

<a href="Chado" class="mw-redirect" title="Chado">Chado</a> may or may
not support multiple alignments.




[Categories](Special%3ACategories "Special%3ACategories"):

- [ApiDB](Category%3AApiDB "Category%3AApiDB")
- [Apollo](Category%3AApollo "Category%3AApollo")
- [Community
  Annotation](Category%3ACommunity_Annotation "Category%3ACommunity Annotation")
- [GBrowse](Category%3AGBrowse "Category%3AGBrowse")
- [GBrowse syn](Category%3AGBrowse_syn "Category%3AGBrowse syn")
- [GMODWeb](Category%3AGMODWeb "Category%3AGMODWeb")
- [Meetings](Category%3AMeetings "Category%3AMeetings")
- [ParameciumDB](Category%3AParameciumDB "Category%3AParameciumDB")
- [Turnkey](Category%3ATurnkey "Category%3ATurnkey")
- [SGN](Category%3ASGN "Category%3ASGN")
- [JBrowse](Category%3AJBrowse "Category%3AJBrowse")
- [DAS](Category%3ADAS "Category%3ADAS")






## Navigation menu









### Navigation



- <span id="n-GMOD-Home">[GMOD Home](Main_Page)</span>
- <span id="n-Software">[Software](GMOD_Components)</span>
- <span id="n-Categories-.2F-Tags">[Categories /
  Tags](Categories)</span>




### Documentation



- <span id="n-Overview">[Overview](Overview)</span>
- <span id="n-FAQs">[FAQs](Category%3AFAQ)</span>
- <span id="n-HOWTOs">[HOWTOs](Category%3AHOWTO)</span>
- <span id="n-Glossary">[Glossary](Glossary)</span>




### Community



- <span id="n-GMOD-News">[GMOD News](GMOD_News)</span>
- <span id="n-Training-.2F-Outreach">[Training /
  Outreach](Training_and_Outreach)</span>
- <span id="n-Support">[Support](Support)</span>
- <span id="n-GMOD-Promotion">[GMOD Promotion](GMOD_Promotion)</span>
- <span id="n-Meetings">[Meetings](Meetings)</span>
- <span id="n-Calendar">[Calendar](Calendar)</span>




### Tools

- <span id="t-smwbrowselink"><a href="Special%3ABrowse/November_2007_GMOD_Meeting"
  rel="smw-browse">Browse properties</a></span>



- <span id="footer-info-lastmod">Last updated at 05:54 on 5 January
  2011.</span>
<!-- - <span id="footer-info-viewcount">359,101 page views.</span> -->
- <span id="footer-info-copyright">Content is available under
  <a href="http://www.gnu.org/licenses/fdl-1.3.html" class="external"
  rel="nofollow">a GNU Free Documentation License</a> unless otherwise
  noted.</span>

<!-- -->



<!-- -->




